fastapi/openapi/utils.py

```diff
@@ -37,7 +37,11 @@
     "title": "ValidationError",
     "type": "object",
     "properties": {
-        "loc": {"title": "Location", "type": "array", "items": {"type": "string"}},
+        "loc": {
+            "title": "Location",
+            "type": "array",
+            "items": {"oneOf": [{"type": "string"}, {"type": "integer"}]},
```

Comment:
I'd suggest using `anyOf` instead (as Open API 3.0 doesn't support `type` as arrays) - it is slightly faster on the validation side. It doesn't require validating the input instance against all sub-schemas + these types are mutually exclusive (e.g. `number` vs `integer` might give different results under `anyOf` and `oneOf`).

Reply:
Btw, the "Dog vs. Cat" example on that page is wrong :) Neither schema enforces `required` properties or restricts extra ones (via `additionalProperties: false`). E.g. `{"bark": true, "breed": "Dingo"}` is a valid `Cat` instance too!